app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @review.user = current_user
+    if @review.save
+      redirect_to action: 'show', controller:'films', id:params[:review][:film_id].to_i
+    else
+      render action: 'show', controller:'films', id:params[:review][:film_id].to_i
```

Comment:
I think this might crash - films show expects there to be an @film in the view context - which doesn't exist here.